include/ignition/gazebo/comms/Broker.hh

```diff
@@ -0,0 +1,154 @@
+  ///    <bind_service>: Service name used to bind an address.
+  ///                    The default value is "/broker/bind"
+  ///    <unbind_service>: Service name used to unbind from an address.
+  ///                      The default value is "/broker/unbind"
+  ///
+  /// Here's an example:
+  /// <plugin
+  ///   filename="ignition-gazebo-perfect-comms-system"
+  ///   name="ignition::gazebo::systems::PerfectComms">
+  ///   <broker>
+  ///     <messages_topic>/broker/inbound</messages_topic>
+  ///     <bind_service>/broker/bind_address</bind_service>
+  ///     <unbind_service>/broker/unbind_address</unbind_service>
+  ///   </broker>
+  /// </plugin>
+  class Broker
+  {
+    /// \brief Constructor.
+    public: Broker();
+
+    /// \brief Destructor.
+    public: virtual ~Broker();
+
+    /// \brief Configure the broker via SDF.
+    /// \param[in] _sdf The SDF Element associated with the broker parameters.
+    public: void Load(std::shared_ptr<const sdf::Element> _sdf);
+
+    /// \brief Start handling comms services.
+    ///
+    /// This function allows us to wait to advertise capabilities to
+    /// clients until the broker has been entirely initialized.
+    public: void Start();
+
+    /// \brief Get the current time.
+    /// \return Current time.
+    public: std::chrono::steady_clock::duration Time() const;
+
+    /// \brief Set the current time.
+    /// \param[in] _time Current time.
+    public: void SetTime(const std::chrono::steady_clock::duration &_time);
+
+    /// \brief This method associates an address with a client topic used as
+    /// callback for receiving messages. This is a client requirement to
+    /// start receiving messages.
+    /// \param[in] _req Bind request containing the following content:
+    ///   _req[0] Client address.
+    ///   _req[1] Model name associated to the address.
+    ///   _req[2] Client subscription topic.
+    /// \param[out] _rep Unused
+    /// \return True when the bind service succeeded or false otherwise.
+    public: bool OnBind(const ignition::msgs::StringMsg_V &_req,
+                        ignition::msgs::Boolean &_rep);
+
+    /// \brief Unbind a given client address. The client associated to this
+    /// address will not receive any more messages.
+    /// \param[in] _req Bind request containing the following content:
+    ///   _req[0] Client address.
+    ///   _req[1] Client subscription topic.
+    public: void OnUnbind(const ignition::msgs::StringMsg_V &_req);
+
+    /// \brief Callback executed to process a communication request from one of
+    /// the clients.
+    /// \param[in] _msg The message from the client.
+    public: void OnMsg(const ignition::msgs::Dataframe &_msg);
+
+    /// \brief Process all the messages in the inbound queue and deliver them
+    /// to the destination clients.
+    public: void DeliverMsgs();
+
+    /// \brief Get a mutable reference to the message manager.
+    public: MsgManager &DataManager();
```

Comment:
```suggestion
    /// \return 
    public: MsgManager &DataManager();
```

Reply:
Added in [beefb6b](https://github.com/ignitionrobotics/ign-gazebo/pull/1416/commits/beefb6bf950dc48ffb783c2c17ca2357ca8ad846).